public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx

```diff
@@ -39,6 +45,38 @@ export class TimePickerSettings extends PureComponent<Props, State> {
     this.setState({ isNowDelayValid: false });
   };
 
+  onMaxTimeRangeChange = (event: React.FormEvent<HTMLInputElement>) => {
+    const value = event.currentTarget.value;
+
+    if (isEmpty(value)) {
+      this.setState({ isMaxTimeRangeValid: true });
+      return this.props.onMaxTimeRangeChange(value);
+    }
+
+    if (value.match(/^[1-9]\d*[yMwdhms]$/)) {
```

Comment:
I think this should be broken out into a validate interval function moved to https://github.com/grafana/grafana/blob/c02ead911345516bc2a4621e86da0a22efe7c37d/packages/grafana-data/src/datetime/rangeutil.ts#L279 to use the same regex as when parsing the interval.

Reply:
I have just modified the regex and I include the millis(ms). Could you please explain what is wrong with my regex?

Reply:
The difference between two regular expressions is that my regex always discards the interval starting with zero(01h, 098m, etc).

Reply:
The interval_regex from rangeutils.ts is what's going to be used when the saved interval value [is parsed](https://github.com/grafana/grafana/blob/c02ead911345516bc2a4621e86da0a22efe7c37d/packages/grafana-data/src/datetime/rangeutil.ts#L302). I think it makes sense to use the same regex to validate the input.

Reply:
Ok, thanks. I understood. I will modify this.